src/api/admin/controller/admin.ts

```diff
@@ -1,29 +1,73 @@
 import { NextFunction, Request, Response } from 'express';
 import moment from 'moment';
 
-import Agency from '../../db/models/Agency';
-import User from '../../db/models/User';
-import AgencyRepository from '../../db/repository/AgencyRepository';
-import UserRepository from '../../db/repository/UserRepository';
-import Messaging from '../../helper/messaging';
-
-import BaseController from './basecontroller';
+import Agency from '../../../db/models/Agency';
+import User from '../../../db/models/User';
+import AgencyRepository from '../../../db/repository/AgencyRepository';
+import UserRepository from '../../../db/repository/UserRepository';
+import WishCardRepository from '../../../db/repository/WishCardRepository';
+import Messaging from '../../../helper/messaging';
+import BaseController from '../../controller/basecontroller';
 
 export default class AdminController extends BaseController {
 	private agencyRepository: AgencyRepository;
-
 	private userRepository: UserRepository;
+	private wishCardRepository: WishCardRepository;
 
 	constructor() {
 		super();
 
 		this.agencyRepository = new AgencyRepository();
 		this.userRepository = new UserRepository();
+		this.wishCardRepository = new WishCardRepository();
 
 		this.handleGetAgencyOverview = this.handleGetAgencyOverview.bind(this);
 		this.handleGetAgencyDetail = this.handleGetAgencyDetail.bind(this);
 		this.handleVerifyAgency = this.handleVerifyAgency.bind(this);
 		this.handleUpdateAgencyData = this.handleUpdateAgencyData.bind(this);
+		this.handleGetDraftWishcards = this.handleGetDraftWishcards.bind(this);
+		this.handlePutDraftWishcard = this.handlePutDraftWishcard.bind(this);
+	}
+
+	async handleGetDraftWishcards(_req: Request, res: Response, _next: NextFunction) {
+		try {
+			const wishcards = await this.wishCardRepository.getWishCardsByStatus('draft');
+			const agenciesWithWishCardsMap: Map<string, any[]> = new Map();
+
+			for (const wishCard of wishcards) {
+				const agencyName = wishCard.belongsTo.agencyName;
+				if (!agenciesWithWishCardsMap.has(agencyName)) {
+					agenciesWithWishCardsMap.set(agencyName, []);
+				}
+				agenciesWithWishCardsMap.get(agencyName)?.push(wishCard);
+			}
+			const agenciesWithWishCards = Object.fromEntries(agenciesWithWishCardsMap);
+			return res.status(200).send({ success: true, data: agenciesWithWishCards });
```

Comment:
status 200 and success in the body are redundant, instead use this basecontroller function

```js
return this.sendResponse(res, agenciesWithWishCards);